### PR TITLE
Type annotation arrow expr

### PR DESCRIPTION
**Pull request by FBroy**
# Description

Add type annotation to the arrowExpr.

## Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

**Review by HectorPeeters — APPROVED** [review]
This looks really good, I hope all tests pass!

**Review by maanpalle — APPROVED** [review]

**Review by yuanzexiong — APPROVED** [review]
This looks pretty standard and yes one more item off the list awesome!

File: src/typeInference/annotateArrowExpr.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	if (!staticContext) return undefined;
+
+	const func = astHelper.getFirstChild(ast, 'EQName');
+
+	// There may be no name for the function
+	if (!func) {
+		return undefined;
+	}
+
+	// Sometimes there is no prefix given, hence we need to check for that case and give an empty prefix
```
Comment (MTimmerije) [inline src/typeInference/annotateArrowExpr.ts:27]: This could be done cleaner I believe, using astHelper.getQName()

**Review by MTimmerije — COMMENTED** [review]
So it looks good, though I would change that one thing to be cleaner/more consistent/robust.

**Pull request merged**